nifi-extension-bundles/nifi-atlassian-bundle/nifi-atlassian-extensions/src/main/java/org/apache/nifi/atlassian/bitbucket/BitBucketFlowRegistryClient.java

```diff
@@ -0,0 +1,163 @@
+            .name("OAuth2 Access Token Provider")
+            .description("Service providing OAuth2 Access Tokens for authentication")
+            .identifiesControllerService(OAuth2AccessTokenProvider.class)
+            .required(true)
+            .dependsOn(AUTHENTICATION_TYPE, BitBucketAuthenticationType.OAUTH2)
+            .build();
+
+    static final PropertyDescriptor WEBCLIENT_SERVICE = new PropertyDescriptor.Builder()
+            .name("Web Client Service")
+            .description("The Web Client Service to use for communicating with BitBucket")
+            .required(true)
+            .identifiesControllerService(WebClientServiceProvider.class)
+            .build();
+
+    static final List<PropertyDescriptor> PROPERTY_DESCRIPTORS = List.of(
+            WEBCLIENT_SERVICE,
+            BITBUCKET_API_URL,
+            BITBUCKET_API_VERSION,
+            WORKSPACE_NAME,
+            REPOSITORY_NAME,
+            AUTHENTICATION_TYPE,
+            ACCESS_TOKEN,
+            USERNAME,
+            APP_PASSWORD,
+            OAUTH_TOKEN_PROVIDER);
+
+    @Override
+    protected List<PropertyDescriptor> createPropertyDescriptors() {
+        return PROPERTY_DESCRIPTORS;
+    }
+
+    @Override
+    protected GitRepositoryClient createRepositoryClient(final FlowRegistryClientConfigurationContext context) throws FlowRegistryException {
+        return BitBucketRepositoryClient.builder()
+                .clientId(getIdentifier())
+                .apiUrl(context.getProperty(BITBUCKET_API_URL).getValue())
+                .apiVersion(context.getProperty(BITBUCKET_API_VERSION).getValue())
+                .workspace(context.getProperty(WORKSPACE_NAME).getValue())
+                .repoName(context.getProperty(REPOSITORY_NAME).getValue())
+                .repoPath(context.getProperty(REPOSITORY_PATH).getValue())
+                .authenticationType(context.getProperty(AUTHENTICATION_TYPE).asAllowableValue(BitBucketAuthenticationType.class))
+                .accessToken(context.getProperty(ACCESS_TOKEN).evaluateAttributeExpressions().getValue())
+                .username(context.getProperty(USERNAME).evaluateAttributeExpressions().getValue())
+                .appPassword(context.getProperty(APP_PASSWORD).evaluateAttributeExpressions().getValue())
+                .oauthService(context.getProperty(OAUTH_TOKEN_PROVIDER).asControllerService(OAuth2AccessTokenProvider.class))
+                .webClient(context.getProperty(WEBCLIENT_SERVICE).asControllerService(WebClientServiceProvider.class))
+                .build();
+    }
+
+    @Override
+    public boolean isStorageLocationApplicable(FlowRegistryClientConfigurationContext context, String location) {
+        // TODO Auto-generated method stub
+        return false;
+    }
+
+    @Override
+    protected String getStorageLocation(GitRepositoryClient repositoryClient) {
+        // TODO Auto-generated method stub
+        return null;
+    }
```

Comment:
Technically it shouldn't cause any problems to not implement these, but the pattern we used for GitHub and GitLab was to construct a string that is similar to the value you use when cloning the repo:
```
git@github.com:<owner>/<repo>.git
```
And then `isStorageLocationApplicable` checks if it starts with the prefix `git@github.com:`.

Reply:
Thanks @bbende - I completely forgot to get back to this when I switched to the other class and testing... Will address now